### ISSUE TITLE
Plans for umpacher to support unity 2021.x versions? 

### ISSUE THREAD

**Issue by Neoshrimp** [dnSpyEx/dnSpy#174]
I'm trying to build patched mono for 2021.3.18.

`./umpatcher.exe 2021.3.18f1-mbe 81a7696b7c1960113bebfe610ac3e693c7d41fce mono/ dnSpyEx-Unity-mono/`

`dnSpy-Unity-mono-v2021.x-V40.sln doesn't exist` error pops up, similar to #123 

However, even with solution file added, git actions and safety checks gutted from the umpacher, errors relating to the patching process emerge, for example, `'msvc\libgcbdwgc.vcxproj' doesn't exist` or `Couldn't patch project ToolsVersion attribute, file 'msvc\build-init.vcxproj'`. Does that mean that the structure of mono project has changed in a way which can't be handled by the current umpacher? Or perhaps it's some error on my part.

Besides, the alternative method of attaching debugger to debug build of the game failed, so I'm guessing building mono is the only avenue left if I want to make use of dnspy's debugger.

**Comment by sunnamed434** [comment]
Hi!

Try to use the alternative [way](https://github.com/dnSpyEx/dnSpy/issues/123#issuecomment-1304597667), I'm not sure if you want to just debug it

**Comment by ElektroKill** [comment]
> However, even with solution file added, git actions and safety checks gutted from the umpacher, errors relating to the patching process emerge, for example, `'msvc\libgcbdwgc.vcxproj' doesn't exist` or `Couldn't patch project ToolsVersion attribute, file 'msvc\build-init.vcxproj'`. Does that mean that the structure of mono project has changed in a way which can't be handled by the current umpacher? Or perhaps it's some error on my part.

This probably means that the project files underwent some changes and the `umpatcher` application is unable to patch them. In order to get this working you would need to identify these breaking changes and update `umpatcher` accordingly. I myself do not have much knowledge when it comes to Mono and especially Unity's use of it so I'm sorry to say that you are kind of on your own here. The `dnSpy-Unity-mono` repository is currently in a contributor-only update state.

**Comment by c0desl1ng3r** [comment]
Amazing work, @Neoshrimp! Any chance of cutting a release for 2021.3.14f1? (the instructions are a bit over my head)

**Comment by Neoshrimp** [comment]
@c0desl1ng3r sure. Added at `builds/`. Although can't guarantee 100% it's gonna work. And also you might have an easier time running a debug server through doorstop.

**Comment by c0desl1ng3r** [comment]
Legend - thankyou so much 🙏

**Comment by war3i4i** [comment]
@Neoshrimp Sorry for asking sir but if you have spare time could you please build 2020.3.33f debug mono-2.0-bdwgc.dll? The instruction is too hard to follow for me, But it seems like dnspyex has files for this particular version
